sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1294,32 +1294,34 @@ def test_large_regularization(penalty):
 
 def test_tol_parameter():
     # Test that the tol parameter behaves as expected
-    X = StandardScaler().fit_transform(iris.data)
-    y = iris.target == 1
-
-    # With tol is None, the number of iteration should be equal to max_iter
-    max_iter = 42
-    model_0 = SGDClassifier(tol=None, random_state=0, max_iter=max_iter)
-    model_0.fit(X, y)
-    assert_equal(max_iter, model_0.n_iter_)
-
-    # If tol is not None, the number of iteration should be less than max_iter
-    max_iter = 2000
-    model_1 = SGDClassifier(tol=0, random_state=0, max_iter=max_iter)
-    model_1.fit(X, y)
-    assert_greater(max_iter, model_1.n_iter_)
-    assert_greater(model_1.n_iter_, 5)
-
-    # A larger tol should yield a smaller number of iteration
-    model_2 = SGDClassifier(tol=0.1, random_state=0, max_iter=max_iter)
-    model_2.fit(X, y)
-    assert_greater(model_1.n_iter_, model_2.n_iter_)
-    assert_greater(model_2.n_iter_, 3)
-
-    # Strict tolerance and small max_iter should trigger a warning
-    model_3 = SGDClassifier(max_iter=3, tol=1e-3, random_state=0)
-    model_3 = assert_warns(ConvergenceWarning, model_3.fit, X, y)
-    assert_equal(model_3.n_iter_, 3)
+    X, y = datasets.make_classification(n_samples=1000, n_features=100,
+                                        n_informative=5, n_classes=2,
+                                        random_state=1234)
+    for seed in range(10):
```

Comment:
10 seeds, that's a bit brutal.

What's the time of this test? Is it very fast? We really need to be careful with the run-time of our test suite.